GAE/src/org/akvo/flow/servlet/ReportServlet.java

```diff
@@ -0,0 +1,237 @@
+import java.util.Date;
+import java.util.logging.Logger;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.akvo.flow.dao.ReportDao;
+import org.akvo.flow.domain.persistent.Report;
+import org.akvo.flow.rest.dto.ReportTaskRequest;
+import org.codehaus.jackson.JsonGenerationException;
+import org.codehaus.jackson.map.JsonMappingException;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.jfree.util.Log;
+import org.waterforpeople.mapping.app.web.dto.TaskRequest;
+
+import com.gallatinsystems.common.Constants;
+import com.gallatinsystems.common.util.PropertyUtil;
+import com.gallatinsystems.framework.rest.AbstractRestApiServlet;
+import com.gallatinsystems.framework.rest.RestRequest;
+import com.gallatinsystems.framework.rest.RestResponse;
+import com.gallatinsystems.user.dao.UserDao;
+import com.google.appengine.api.taskqueue.Queue;
+import com.google.appengine.api.taskqueue.QueueFactory;
+import com.google.appengine.api.taskqueue.TaskOptions;
+
+
+public class ReportServlet extends AbstractRestApiServlet {
+    private static final Logger log = Logger.getLogger(ReportServlet.class.getName());
+
+    private static final long serialVersionUID = -9064136799930675167L;
+    private static final String SERVLET_URL = "/app_worker/reportservlet";
+
+    private ReportDao rDao;
+    private UserDao uDao;
+
+    class ReportOptions implements Serializable {
+        /**
+         *
+         */
+        private static final long serialVersionUID = 1L;
+        public String exportMode;
+        public Long reportId;
+        public Long questionId; //only for GeoJSON
+        public Date from;
+        public Date to;
+        public Boolean lastCollection;
+        public String imgPrefix;
+        public String uploadUrl;
+    }
+
+    class ReportCriteria implements Serializable {
+        /**
+         *
+         */
+        private static final long serialVersionUID = 1L;
+        public ReportOptions opts;
+        public String exportType;
+        public String appId;
+        public Long surveyId;
+        public String email;
+    }
+
+    class ReportBody implements Serializable {
+        /**
+         *
+         */
+        private static final long serialVersionUID = 1L;
+        public ReportCriteria criteria;
+    }
+
+    public ReportServlet() {
+        rDao = new ReportDao();
+        uDao = new UserDao();
+    }
+
+    @Override
+    protected RestRequest convertRequest() throws Exception {
+        HttpServletRequest req = getRequest();
+        RestRequest restRequest = new ReportTaskRequest();
+        restRequest.populateFromHttpRequest(req);
+        return restRequest;
+    }
+
+    @Override
+    protected RestResponse handleRequest(RestRequest req) throws Exception  {
+        ReportTaskRequest stReq = (ReportTaskRequest) req;
+        String action = stReq.getAction();
+        Long id = stReq.getId();
+        log.info("action: " + action + " id: " + id);
+        if (action == null) {
```

Comment:
should this ever happen given there is a check for a missing action parameter when initialising the `ReportTaskRequest` object?

Reply:
Probably not.